### ISSUE TITLE
Do not add transports to default extensions

### ISSUE THREAD

**Issue by Kagami** [telehash/telehash-js#23]
By running this simple mesh node creates 3 listening sockets: two for TCP and one for UDP:

``` js
var th = require("telehash");

th.generate(function(err, id) {
  th.mesh({id: id}, function(err, mesh) {
    console.log(mesh);
  });
});
```

That's because of the transport extensions [added by default](https://github.com/telehash/telehash-js/blob/master/index.js#L8) on module load.

I could unload them by calling:

``` js
delete th.extensions.udp4;
delete th.extensions.tcp4;
delete th.extensions.http;
```

but it looks ugly as for me. (Also little bug: [extensions](https://github.com/telehash/telehash-js/blob/master/lib/mesh.js#L27) should be an object.)
Should it be the option to disable default extensions without manually removing them?

**Comment by bachp** [comment]
I think it would make sense to have something like [`app.use`](http://expressjs.com/api.html#app.use) from [express.js](http://expressjs.com/) where the user can register all required extension.

Only the registered extensions should be available. Same for ciphersets and other things.
This would also help in some deployment scenarions (sodium isn't running everywhere).

**Comment by quartzjer** [comment]
Yes, I very much want to move to this model here too, it's near the top of my TODO list now so it should happen soon :)
